syntax = "proto3";

//请求释放技能
message SkillCastReq
{	
	uint64	ObjectGuid	= 1;
	int32	SkillID		= 2;
	float	HostX		= 3;
	float	HostY		= 4;
	float	HostZ		= 5;
	float	HostFt		= 6;
	float	TargetX		= 7;
	float	TargetY		= 8;
	float	TargetZ		= 9;
	float	TargetFt	= 10;
	repeated	uint64	TargetObjects = 11;
}

//请求释放技能返回(如要释放是允许则没有这个返回)
message SkillCastAck
{	
	uint32	RetCode = 1;  //是否释放成功, 失败原因。
}

//技能释放结果
message SkillResultItem
{	
	uint64	ObjectGuid	= 1;	//目标的ID
	int32	SkillID		= 2;	//被哪个技能击中了
	int32   ActionID    = 3;	//现在需要什么动作
	float	OrgX		= 4;	//被击中时的坐标
	float	OrgY		= 5;	//被击中时的坐标
	float	OrgZ		= 6;	//被击中时的坐标
	float	OrgFt		= 7;	//被击中时的朝向
	int32   HpChange    = 8;	//变化的血量
	bool	bCrit		= 9;    //是否爆击
}

//技能释放结果
message SkillResultNtf
{	
	repeated SkillResultItem ItemList = 1;
}

//移动动作请求
message ObjectActionReq {
   repeated	ActionReqItem ActionList = 1;
}

//移动动作请求
message ObjectActionNty {
   repeated ActionNtyItem ActionList = 1;
}

message ActionReqItem {
	uint64	ObjectGuid	= 1;
	int32	ActionID	= 2;
	float	HostX		= 3;
	float	HostY		= 4;
	float	HostZ		= 5;
	float	HostFt		= 6;
}


message ObjectActionAck {
   repeated	ActionReqItem ActionList = 1;
}

message ActionNtyItem {
	uint64	ObjectGuid	= 1;
	int32	ActionID	= 2;
	int32	Camp		= 3;
	int32	ObjectState	= 4;		//实例状态
	int32	BuffState	= 5;		//特效状态
	float	HostX		= 6;
	float	HostY		= 7;
	float	HostZ		= 8;
	float	HostFt		= 9;
	int32	ActorID		= 10;		//Actor ID
	int32	Level		= 11;		//等级
	uint64	ControlerID	= 12;     	//AI控制人的GUID
	int32	Hp			= 13;
	int32	Mp			= 14;
	int32	HpMax		= 15;
	int32	MpMax		= 16;
}

message NewItem {
	uint64	ObjectGuid	= 1;		//实例ID
	int32	ObjType		= 2;		//对象类别
	int32	ObjectState	= 3;		//实例状态
	int32	BuffState	= 4;		//特效状态
	int32	ActorID		= 5;		//Actor ID
	int32	Camp		= 6;
	string	Name		= 7;		//名字
	int32	Level		= 8;		//等级
	uint64	HostGuid	= 9;		//主人的GUID
	uint64	ControlerID	= 10;     	//AI控制人的GUID
	uint64	SummonID 	= 11;		//招唤者的GUID
	float	X			= 12;		
	float	Y			= 13;
	float	Z			= 14;
	float	Ft			= 15;
	int32	Hp			= 16;
	int32	Mp			= 17;
	int32	HpMax		= 18;
	int32	MpMax		= 19;
	int32	Weapon		= 20; 		//武器
}

message ObjectNewNty {
  repeated NewItem NewList = 1;
}

message ObjectRemoveNty {
  repeated uint64 RemoveList = 1;
}

//子弹对象
message BulletItem
{	
	uint64	ObjectGuid	= 1;	//对象ID
	int32	BulletID	= 2;	//子弹的ID
	float	X			= 3;	//坐标X
	float	Y			= 4;	//坐标Y
	float	Z			= 5;	//坐标X
	float	VX			= 6;	//方向X
	float	VZ			= 7;	//方向Y
	float   Speed		= 8;	//速度
	float   AccSpeed	= 9;    //加速度
	float   LeftTime	= 10;   //剩余时间
}

//技能释放结果
message BulletNewNtf
{	
	repeated BulletItem ItemList = 1;
}

message HeartBeatReq {
  uint32  TimeStamp = 1;
}

message HeartBeatAck {
  uint32  TimeStamp = 1;
  uint32  ServerTime = 2;
}

message ObjectDieNotify {
  uint64  ObjectGuid = 1;
}
